test/EFCore.PG.FunctionalTests/Query/MathQueryTest.cs

```diff
@@ -0,0 +1,496 @@
+﻿using System;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.TestUtilities;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Npgsql.EntityFrameworkCore.PostgreSQL.TestUtilities;
+using Xunit;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query
+{
+    /// <summary>
+    /// Provides unit tests for math functions.
+    /// </summary>
+    /// <remarks>
+    /// See:
+    ///   - https://www.postgresql.org/docs/current/static/functions-math.html
+    ///   - https://www.postgresql.org/docs/current/static/functions-conditional.html#FUNCTIONS-GREATEST-LEAST
+    /// </remarks>
+    public class MathQueryTest : IClassFixture<MathQueryTest.MathQueryNpgsqlFixture>
```

Comment:
Tests for Math functions already live in SimpleQueryTestBase.Functions.cs, but I understand that you want to test different CLR types and that fixture/context doesn't allow for that.

Another thought is whether xunit supports generic test theories - I know nunit does. It would cut down on lots of code in this test suite.

Reply:
I think xunit does... let me take a look.

Reply:
Oh...well that doesn't seem to work actually, since none of the methods are generic.

It might be possible to put together some theory data that also injects an `Expression<Func<object,object,object>>` or something similar (in place of calling `Math.Min` or `Math.Max` directly), but that seems like more trouble than it's worth.

Reply:
OK